### PR TITLE
TINY-10174: Allow dialog collection items to use icons from icon pack

### PR DESCRIPTION
**Pull request by MitchC1999**
Related Ticket: TINY-10174

Description of Changes:
* Collection item icons are now retrieved from the icon pack before using the string as a fallback
* This will not affect charmap or emoticons dialogs, as those characters are not registered as icons

Pre-checks:
* [x] Changelog entry added
* [x] Tests have been added (if applicable)
* [x] Branch prefixed with `feature/`, `hotfix/` or `spike/`

Review:
* [x] Milestone set
* [x] Docs ticket created (if applicable)

GitHub issues (if applicable):

### CODE REVIEW

**Review by spocke — APPROVED** [review]

File: modules/tinymce/src/themes/silver/main/ts/ui/dialog/Collection.ts [hunk 1]
```diff
@@ -29,6 +29,11 @@ export const renderCollection = (
   // DUPE with TextField.
   const pLabel = spec.label.map((label) => renderLabel(label, providersBackstage));
 
+  const icons = providersBackstage.icons();
+
+  // icon string is either in icon pack or displayed directly
```
Comment (mkzhx) [inline modules/tinymce/src/themes/silver/main/ts/ui/dialog/Collection.ts:34]:
```suggestion
  // TINY-10174: Icon string is either in icon pack or displayed directly
```

File: modules/tinymce/src/themes/silver/test/ts/browser/skin/OxideCollectionComponentTest.ts [hunk 2]
```diff
@@ -6,105 +6,102 @@ import { Attribute, SugarBody, SugarDocument, SugarElement } from '@ephox/sugar'
...
 
-    before(async () => {
-      const editor = hook.editor();
-      TinyUiActions.clickOnToolbar(editor, 'button');
-      await TinyUiActions.pWaitForDialog(editor);
+    before(function () {
+      this.skip();
+      // const editor = hook.editor();
+      // TinyUiActions.clickOnToolbar(editor, 'button');
+      // await TinyUiActions.pWaitForDialog(editor);
```
Comment (mkzhx) [inline modules/tinymce/src/themes/silver/test/ts/browser/skin/OxideCollectionComponentTest.ts:104]: Should this be removed?

Comment (ltrouton) [inline modules/tinymce/src/themes/silver/test/ts/browser/skin/OxideCollectionComponentTest.ts:104]: Not sure what is happening here

**Review by mkzhx — APPROVED** [review]

File: modules/tinymce/src/themes/silver/test/ts/browser/skin/OxideCollectionComponentTest.ts (hunk 2 above)
Comment (MitchC1999) [reply modules/tinymce/src/themes/silver/test/ts/browser/skin/OxideCollectionComponentTest.ts:104]: These tests were previously skipped as they are mostly testing the `columns` property, I just moved them in this PR.

Comment (MitchC1999) [reply modules/tinymce/src/themes/silver/test/ts/browser/skin/OxideCollectionComponentTest.ts:104]: These tests were previously skipped as they are mostly testing the `columns` property, I just moved them in this PR.

Comment (ltrouton) [inline modules/tinymce/src/themes/silver/test/ts/browser/skin/OxideCollectionComponentTest.ts:104]: Ah right, I missed that it was in the existing file. Although it looks some new comments were added which might have been what confused me
```suggestion
			// TODO: TINY-3229 implement collection columns properly
      this.skip();
```

**Review by ltrouton — APPROVED** [review]

**Pull request merged**